#! /usr/bin/env python

PACKAGE='color_segmenter'
import roslib; roslib.load_manifest(PACKAGE)

from math import pi

from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
#Name            Type   Reconfiguration level             Description         Default Min Max
gen.add("red_h_min", int_t, SensorLevels.RECONFIGURE_RUNNING,"Minimum Red Hue Level.",75, 0, 255)
gen.add("red_h_max", int_t, SensorLevels.RECONFIGURE_RUNNING,"Maximum Red Hue Level.",220,0, 255)
gen.add("red_s_min", int_t, SensorLevels.RECONFIGURE_RUNNING,"Minimum Red Saturation Level.",0, 0, 255)
gen.add("red_s_max", int_t, SensorLevels.RECONFIGURE_RUNNING,"Maximum Red Saturation Level.",255, 0, 255)

gen.add("orange_h_min", int_t, SensorLevels.RECONFIGURE_RUNNING,"Minimum Orange Hue Level.",75, 0, 255)
gen.add("orange_h_max", int_t, SensorLevels.RECONFIGURE_RUNNING,"Maximum Orange Hue Level.",220,0, 255)
gen.add("orange_s_min", int_t, SensorLevels.RECONFIGURE_RUNNING,"Minimum Orange Saturation Level.",0, 0, 255)
gen.add("orange_s_max", int_t, SensorLevels.RECONFIGURE_RUNNING,"Maximum Orange Saturation Level.",255, 0, 255)

gen.add("yellow_h_min", int_t, SensorLevels.RECONFIGURE_RUNNING,"Minimum Yellow Hue Level.",75, 0, 255)
gen.add("yellow_h_max", int_t, SensorLevels.RECONFIGURE_RUNNING,"Maximum Yellow Hue Level.",220,0, 255)
gen.add("yellow_s_min", int_t, SensorLevels.RECONFIGURE_RUNNING,"Minimum Yellow Saturation Level.",0, 0, 255)
gen.add("yellow_s_max", int_t, SensorLevels.RECONFIGURE_RUNNING,"Maximum Yellow Saturation Level.",255, 0, 255)

gen.add("green_h_min", int_t, SensorLevels.RECONFIGURE_RUNNING,"Minimum Green Hue Level.",75, 0, 255)
gen.add("green_h_max", int_t, SensorLevels.RECONFIGURE_RUNNING,"Maximum Green Hue Level.",220,0, 255)
gen.add("green_s_min", int_t, SensorLevels.RECONFIGURE_RUNNING,"Minimum Green Saturation Level.",0, 0, 255)
gen.add("green_s_max", int_t, SensorLevels.RECONFIGURE_RUNNING,"Maximum Green Saturation Level.",255, 0, 255)

gen.add("blue_h_min", int_t, SensorLevels.RECONFIGURE_RUNNING,"Minimum Blue Hue Level.",75, 0, 255)
gen.add("blue_h_max", int_t, SensorLevels.RECONFIGURE_RUNNING,"Maximum Blue Hue Level.",220,0, 255)
gen.add("blue_s_min", int_t, SensorLevels.RECONFIGURE_RUNNING,"Minimum Blue Saturation Level.",0, 0, 255)
gen.add("blue_s_max", int_t, SensorLevels.RECONFIGURE_RUNNING,"Maximum Blue Saturation Level.",255, 0, 255)

exit(gen.generate(PACKAGE, "dynamic_reconfigure_node", "ColorSegmenter"))
